Dockerfile

```diff
@@ -41,36 +41,34 @@ RUN apt-get clean \
   && apt-get update \
   && apt-get install -y \
   build-essential python3.9 python3-pip git cmake make ruby rake autoconf automake \
-  && pip install conan -v "conan==1.59.0"
+  && pip install conan -v "conan==2.0.9"
 
 # make an agent directory and cd into it
 WORKDIR /root/agent
 
 # bring in the repo contents, minus .dockerignore files
 COPY . .
 
-# set some variables
-ENV PATH=$HOME/venv3.9/bin:$PATH
-ENV CONAN_PROFILE=conan/profiles/docker
-ENV WITH_RUBY=True
-ENV WITH_TESTS=False
+ARG WITH_RUBY=True
+ARG WITH_TESTS=False
```

Comment:
@wsobel, do we really need these variables to be defined in `Dockerfile`? Wouldn’t it be better to move them to `conan/profiles/docker`?

Reply:
> [@wsobel: [src](https://github.com/mtconnect/cppagent/pull/303#issuecomment-1666924310)]
>
> This is if you want docker build to build and run tests. I have not hooked this up, need a conditional to run create with `-tr “”` or not. I have not figured out the best way to do it.

Well, whether we should do the building and testing the agent in Docker build or not is a good question. It makes the image building longer, howerver, it might be better to do it as separate steps via Google Actions (`build` → `test` → `build_image` → `publish`); in that case, it might be better to as much as possible outside of the `Dockerfile`.

> I’m not sure if you can change the -tr setting in the docker profile. I don’t think so.

I’ll leave the decision of the support of `-tr` to you.

Reply:
A little background, with conan 2 and the best practice [conan create](https://docs.conan.io/2/reference/commands/create.html?highlight=test_package) is to have a separate project in the `test_package` folder. The test folder is set using -tr and using -tr "" disables the building and running of this package. 

I'll put a conditional expression in the Dockerfile to evaluate first to set the values for -tr and `skip_tests` builtin build option. Just haven't got around to this yet. Working on automating the deploy now.

Reply:
> I'll put a conditional expression in the Dockerfile to evaluate first to set the values for -tr and `skip_tests` builtin build option.

Do you mean to use a `--build-arg skip_tests='True'` when building the image?

Reply:
I was referring to the conan configuration: `-c build.tools:skip_tests=True`

See: [conan global.conf](https://docs.conan.io/2/reference/config_files/global_conf.html?highlight=skip_tests)